README.md

```diff
@@ -1,2 +1,118 @@
+    - `file` - path to Exercise markup file, relative to `.strigo` directory, supported extensions `.md` (markdown) and `.rst` (restructured), any other extension should be overriden by `syntax` field (see below)
+    - `title` - defines Exercise title in Strigo interface
+    - `syntax` - overrides `file` extension, supported syntaxes `.md` (markdown) and `.rst` (restructured)
+
+---
+
+## Main concepts
+
+1. **Hello world**
+
+   Just configure the order of the files, set titles and it is good to go.
+
+   ```yaml
+   ---
+   classes:
+     - exercises:
+         # `file` is relative to .strigo directory
+         - file: "../hello-world/hello.md"
+           title: "Hello!"
+         - file: "../hello-world/world.txt"
+           title: "World!"
+   ```
+
+1. **Multiple Classes in a single repository**
+
+   You can have multiple Classes in a single repository, defining Class IDs for each.
+
+   ```yaml
+   ---
+   classes:
+     - class_id: "<class_id_1>"
+       exercises:
+         - file: "../1.md"
+           title: "Title 1!"
+         - file: "../2.md"
+           title: "Title 2!"
+     - class_id: "<class_id_2>"
+       exercises:
+         - file: "../3.md"
+           title: "Title 3!"
+         - file: "../4.md"
+           title: "Title 4!"
+   ```
+
+1. **Reusing Exercises for different classes**
+
+   There is no restriction on using the same file for multiple Classes, or even using the same file multiple times for the same Class
+
+   ```yaml
+   ---
+   classes:
+     - class_id: "<class_id_1>"
+       exercises:
+         - file: "../1.md"
+           title: "Title 1!"
+         - file: "../1.md"
+           title: "Title 1!"
+     - class_id: "<class_id_2>"
+       exercises:
+         - file: "../1.md"
+           title: "Title 1!"
+   ```
+
+1. **Markup syntaxes**
+
+   You can use different markup syntaxes by using `syntax` field for each Exercise
+
+   ```yaml
+   ---
+   classes:
+     - class_id: "<class_id_1>"
+       exercises:
+         - file: "../1.txt"
+           title: "Markdown!"
+           # `syntax` is optional
+           # can be "md" | "rst"
+           # uses file extension as fallback
+           syntax: "md"
+         - file: "../2.txt"
+           title: "Title 1!"
+           syntax: "rst"
+   ```
+
+1. **Relative vs Absolute paths to images**
+
+   Your Exercises' repository can have images stored inside the repository, a.k.a Relative images. Fetching these exercises will upload these images to Strigo. If the image could not be accessed, we will replace it with a placeholder instead.
+
+   Absolute images paths will not be affected, so if the image cannot be accessed, it won't show up for your students.
+
+1. **Public vs Private repository**
+
+   There is no difference between Public and Private repositories configuration structure. Please follow [Strigo Guidelines](http://help.strigo.io/en/articles/4951906-fetching-lab-exercises-from-github) for more details.
```

Comment:
"Both public and private repositories should be configured the same way." ...

"Strigo's Guidelines"